From 097e3212e76d7b694ea7cb43e525391dd641ac09 Mon Sep 17 00:00:00 2001
From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Sun, 24 Apr 2016 08:45:44 +0200
Subject: [PATCH 28/93] rules.mk: make the locked template available even if
 flock has not been built yet (fall back to unlocked shell command)

Signed-off-by: Felix Fietkau <nbd@openwrt.org>

Backport of r48414
---
 rules.mk | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/rules.mk b/rules.mk
index 9d0134d2b2..ece502609b 100644
--- a/rules.mk
+++ b/rules.mk
@@ -310,12 +310,16 @@ endef
 # Execute commands under flock
 # $(1) => The shell expression.
 # $(2) => The lock name. If not given, the global lock will be used.
-define locked
+ifneq ($(wildcard $(STAGING_DIR_HOST)/bin/flock),)
+  define locked
 	SHELL= \
-	$(STAGING_DIR_HOST)/bin/flock \
+	flock \
 		$(TMP_DIR)/.$(if $(2),$(strip $(2)),global).flock \
 		-c '$(subst ','\'',$(1))'
-endef
+  endef
+else
+  locked=$(1)
+endif
 
 # Recursively copy paths into another directory, purge dangling
 # symlinks before.
-- 
2.11.0

